rpmlint/checks/PolkitCheck.py

```diff
@@ -0,0 +1,129 @@
+import os
+from re import split
+from xml.dom.minidom import parse
+from xml.parsers.expat import ExpatError
+
+from rpmlint.checks.AbstractCheck import AbstractCheck
+
+
+class PolkitCheck(AbstractCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.polkit_privs_whitelist = config.configuration.get('PolkitPrivsWhitelist', ())
+        self.polkit_privs_files = config.configuration.get('PolkitPrivsFiles', ['/etc/polkit-default-privs.standard'])
+        self.privs = {}
+        self._collect_privs()
+
+    def _collect_privs(self):
+        for filename in self.polkit_privs_files:
+            if os.path.exists(filename):
+                self._parse_privs_file(filename)
+
+    def _parse_privs_file(self, filename):
+        with open(filename) as inputfile:
+            for line in inputfile:
+                line = line.split('#')[0].rstrip()
+                if line:
+                    line = split(r'\s+', line)
+                    priv = line[0]
+                    value = line[1]
+                    self.privs[priv] = value
+
+    def check_perm_files(self, pkg):
+        """Checks files in polkit-default-privs.d."""
+        prefix = '/etc/polkit-default-privs.d/'
```

Comment:
What is this directory and where does it come from?

Reply:
This directory can be used by applications to write their own polkit profile to `/etc/polkit-default-privs.d/` (such as for example https://github.com/openSUSE/polkit-default-privs/blob/master/profiles/polkit-default-privs.standard). These profile paths have to be whitelisted in the rpmlint config.

The check and the path come from here: https://github.com/openSUSE/rpmlint-checks/blob/master/CheckPolkitPrivs.py#L75

This is currently only used like that by opensuse but other users could theoretically also do it like that.

Reply:
How does this even work? It doesn't seem integrated into polkit itself? Has anyone discussed it with them?